Off-Top-App/off-top-python
# OT-67 | SG - Create Spark Streaming Instance in Python

### What I did:
- hook up spark streaming to our python repo. 
- Keeping it organized by creating a file called SparkConfig.py and creating methods/services(maybe) in SparkMethods.py 

### How To Test:
- **PREREQ:** You will need to download spark and hadoop. Click here: `https://spark.apache.org/downloads.html` and make sure to select Spark release 3.0.0 release and Hadoop 2.7. 
- **PREREQ:** Add spark-submit and spark-shell to your path, if not you can do `<path-of-spark-hadoop>/bin/spark-submit`
- **PREREQ:** You will need to download pip, which you download by using `python install pip`, which means you will also need to download 3.7.+
    - Once you have pip, `cd` into the directory of SparkConfig.py and run: `pip install pyspark`. You might need pip3
- Once you have everything you will need to:
   1. open up a terminal tab and run `nc -lk 9999` as a netcat server that acts as an input client for spark streaming to receive data
   2. Open up another terminal and `cd` into the directory of the off-top-python project and into the directory of /Spark. Then run `spark-submit SparkConfig.py 9999`
  3. Then in the terminal where netcat is running type in a sentence and you should see in the terminal where SparkConfig.py is running the words and the amount of times it pops up. You will see the response below. It will process the input really fast so it would be smart to cancel the spark-submit build right after you submit your text in the netcat client terminal
<img width="651" alt="Screen Shot 2020-02-11 at 12 39 32 AM" src="https://user-images.githubusercontent.com/26909882/74221469-0d7f2400-4c67-11ea-95a0-3e0a6525f4c6.png">
